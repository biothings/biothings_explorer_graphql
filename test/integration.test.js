

const { createTestClient } = require('apollo-server-testing');
const gql = require('graphql-tag');
const getServer = require("../src/server");

describe("integration tests", function () {
  let server;

  beforeAll(async function() {
    server = await getServer();
  });

  beforeEach(function() {
    jest.setTimeout(20000);
  });

  test("Test Disease related_to ChemicalSubstance", async function() {
    const DiseaseRelatedToChemicalSubstance = gql`
      query {
        Disease(ids: "MONDO:0005105") {
          id
          label
          ChemicalSubstance(predicates: related_to, sortBy: ngd_overall, maxResults: 25) {
            id
            label
            api
            publication
            source
            correlation {
              ngd_overall
              ngd_starred
            }
          }
        }
      }
    `
    const { query } = createTestClient(server);
    const res = await query({query: DiseaseRelatedToChemicalSubstance});
    expect(res.data.Disease[0].ChemicalSubstance.length).toEqual(25);
    expect(res).toMatchSnapshot();
  }); 

  test("Test Gene related_to Disease", async function() {
    const GeneRelatedToDisease = gql`
      query {
        Gene(ids: "NCBIGene:285440") {
          id
          label
          Disease(predicates: related_to) {
            id
            label
            api
            publication
          }
        }
      }
    `
    const { query } = createTestClient(server);
    const res = await query({query: GeneRelatedToDisease});
    expect(res).toMatchSnapshot();
  }); 
});