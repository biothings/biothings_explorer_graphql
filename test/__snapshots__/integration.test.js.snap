// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration tests Test Disease related_to ChemicalSubstance 1`] = `
Object {
  "data": Object {
    "Disease": Array [
      Object {
        "ChemicalSubstance": Array [
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.3697514308538006,
              "ngd_starred": 0.4654604652367815,
            },
            "id": "CHEBI:135311",
            "label": "MITOLACTOL",
            "publication": Array [
              "pubmed:3834791",
              "pubmed:4016778",
              "pubmed:6507378",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.42085569090326735,
              "ngd_starred": 0.5037349882100742,
            },
            "id": "CHEBI:30621",
            "label": "ARSENIC TRIOXIDE",
            "publication": Array [
              "pubmed:16130126",
              "pubmed:18337652",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.4355148335876441,
              "ngd_starred": 0.5164943424515374,
            },
            "id": "MESH:C037112",
            "label": "9-deoxy-delta-9-prostaglandin D2",
            "publication": Array [
              "pubmed:8752147",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5014338102743953,
              "ngd_starred": 0.5514635488159519,
            },
            "id": "MESH:D006025",
            "label": "Glycosaminoglycans",
            "publication": Array [
              "pubmed:2094611",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5056602064127781,
              "ngd_starred": 0.6407503046806085,
            },
            "id": "MESH:D003571",
            "label": "Cytochalasin B",
            "publication": Array [
              "pubmed:2302708",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5056602064127781,
              "ngd_starred": 0.6407503046806085,
            },
            "id": "MESH:D034741",
            "label": "Interfering RNA, Small",
            "publication": Array [
              "pubmed:16719778",
              "pubmed:18519701",
              "pubmed:24275138",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "CORD Disease API",
            "correlation": Object {
              "ngd_overall": 0.5056602064127781,
              "ngd_starred": 0.6407503046806085,
            },
            "id": "CHEBI:4305",
            "label": "DACARBAZINE",
            "publication": Array [
              "pmc:PMC3702032",
              "pmc:PMC3816698",
              "pmc:PMC6509811",
              "pmc:PMC6617740",
            ],
            "source": "Translator Text Mining Provider",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5104437073585528,
              "ngd_starred": 0.616455447842857,
            },
            "id": "MESH:C081679",
            "label": "ZCE 025 monoclonal antibody",
            "publication": Array [
              "pubmed:1968968",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5105725182163027,
              "ngd_starred": 0.6476178564082071,
            },
            "id": "CHEBI:138112",
            "label": "Chondroitin",
            "publication": Array [
              "pubmed:12782604",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5165601395298366,
              "ngd_starred": 0.6765120151124993,
            },
            "id": "MESH:D014408",
            "label": "Markers, Tumor",
            "publication": Array [
              "pubmed:10354916",
              "pubmed:10890377",
              "pubmed:11599800",
              "pubmed:12704461",
              "pubmed:1347764",
              "pubmed:14646622",
              "pubmed:15762365",
              "pubmed:15957165",
              "pubmed:17276671",
              "pubmed:18454690",
              "pubmed:18845991",
              "pubmed:1977250",
              "pubmed:21946018",
              "pubmed:23907232",
              "pubmed:24217944",
              "pubmed:2696716",
              "pubmed:4060826",
              "pubmed:6205909",
              "pubmed:6411338",
              "pubmed:8573133",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5329312444264491,
              "ngd_starred": 0.6354596623782988,
            },
            "id": "MESH:D000276",
            "label": "Adjuvant, Immunologic",
            "publication": Array [
              "pubmed:15149168",
              "pubmed:8892054",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5337118242804668,
              "ngd_starred": 0.7616141226183656,
            },
            "id": "UMLS:C1520145",
            "label": "UMLS:C1520145",
            "publication": Array [
              "pubmed:22577889",
              "pubmed:6852975",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5351175390465585,
              "ngd_starred": 0.6082842464232177,
            },
            "id": "CHEBI:16040",
            "label": "CYTOSINE",
            "publication": Array [
              "pubmed:12115485",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5461986821881316,
              "ngd_starred": 0.7250479222609667,
            },
            "id": "MESH:D000939",
            "label": "Epitopes",
            "publication": Array [
              "pubmed:11120859",
              "pubmed:11174375",
              "pubmed:12782604",
              "pubmed:1348919",
              "pubmed:15580499",
              "pubmed:1709658",
              "pubmed:18249038",
              "pubmed:19017974",
              "pubmed:19111423",
              "pubmed:19212346",
              "pubmed:2453068",
              "pubmed:27320826",
              "pubmed:28836878",
              "pubmed:9403705",
              "pubmed:9609139",
              "pubmed:9764809",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.548831514320058,
              "ngd_starred": 0.659309386387933,
            },
            "id": "MESH:D015295",
            "label": "Antigens, Carbohydrate, Tumor-Associated",
            "publication": Array [
              "pubmed:14533809",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5492738509034409,
              "ngd_starred": 0.6130564805110331,
            },
            "id": "MESH:D007657",
            "label": "Bodies, Ketone",
            "publication": Array [
              "pubmed:26869992",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "CORD Disease API",
            "correlation": Object {
              "ngd_overall": 0.5492738509034409,
              "ngd_starred": 0.6130564805110331,
            },
            "id": "CHEBI:23888",
            "label": "CHEBI:23888",
            "publication": Array [
              "pmc:PMC3702032",
              "pmc:PMC6404779",
              "pmc:PMC6505380",
              "pmc:PMC6509811",
              "pmc:PMC6514259",
              "pmc:PMC6520727",
              "pmc:PMC6523645",
              "pmc:PMC6532080",
              "pmc:PMC6617877",
              "pmc:PMC6759318",
            ],
            "source": "Translator Text Mining Provider",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5518011731454999,
              "ngd_starred": 0.6001151234148555,
            },
            "id": "MESH:D006001",
            "label": "Glycoconjugates",
            "publication": Array [
              "pubmed:8098445",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5576203559412253,
              "ngd_starred": 0.6976760813569723,
            },
            "id": "CHEBI:28661",
            "label": "GAMOLENIC ACID",
            "publication": Array [
              "pubmed:1650000",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5604498795709704,
              "ngd_starred": 0.5863167675344533,
            },
            "id": "MESH:D007155",
            "label": "Immunomodulators",
            "publication": Array [
              "pubmed:12681431",
              "pubmed:17893524",
              "pubmed:24050392",
              "pubmed:2531908",
              "pubmed:9798342",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5644796716939003,
              "ngd_starred": 0.6502286554565262,
            },
            "id": "MESH:C000614960",
            "label": "Iodine-125",
            "publication": Array [
              "pubmed:11797294",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5660700174196868,
              "ngd_starred": 0.599151606714789,
            },
            "id": "MESH:D009696",
            "label": "Acids, Nucleic",
            "publication": Array [
              "pubmed:17199564",
              "pubmed:23634303",
              "pubmed:4253736",
              "pubmed:4253738",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5709427414605881,
              "ngd_starred": 0.6759681426557729,
            },
            "id": "UMLS:C0815029",
            "label": "UMLS:C0815029",
            "publication": Array [
              "pubmed:28323387",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "SEMMED Disease API",
            "correlation": Object {
              "ngd_overall": 0.5734127484235313,
              "ngd_starred": 0.6608793682258011,
            },
            "id": "CHEBI:545959",
            "label": "HOMOVANILLIC ACID",
            "publication": Array [
              "pubmed:1106836",
              "pubmed:5987526",
            ],
            "source": "SEMMED",
          },
          Object {
            "api": "CORD Disease API",
            "correlation": Object {
              "ngd_overall": 0.5734160608867758,
              "ngd_starred": 0.9682703486638895,
            },
            "id": "CHEBI:50114",
            "label": "CHEBI:50114",
            "publication": Array [
              "pmc:PMC6572624",
            ],
            "source": "Translator Text Mining Provider",
          },
        ],
        "id": "MONDO:0005105",
        "label": "melanoma (disease)",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`integration tests Test Gene related_to Disease 1`] = `
Object {
  "data": Object {
    "Gene": Array [
      Object {
        "Disease": Array [
          Object {
            "api": "CORD Gene API",
            "id": "MONDO:0001280",
            "label": "choroiditis",
            "publication": Array [
              "pmc:PMC6517852",
            ],
          },
          Object {
            "api": "CORD Gene API",
            "id": "MONDO:0019200",
            "label": "retinitis pigmentosa",
            "publication": Array [
              "pmc:PMC6517852",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "UMLS:C0004093",
            "label": "Asthenia",
            "publication": Array [
              "pubmed:22391096",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "MONDO:0010557",
            "label": "choroideremia",
            "publication": Array [
              "pubmed:27348340",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "MONDO:0003847",
            "label": "inherited genetic disease",
            "publication": Array [
              "pubmed:23302251",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "MONDO:0005044",
            "label": "hypertensive disorder",
            "publication": Array [
              "pubmed:23834962",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "UMLS:C0021368",
            "label": "Inflammation",
            "publication": Array [
              "pubmed:24856391",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "MONDO:0008159",
            "label": "postmenopausal osteoporosis",
            "publication": Array [
              "pubmed:23410587",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "MONDO:0005263",
            "label": "unipolar depression",
            "publication": Array [
              "pubmed:25115444",
              "pubmed:28709058",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "UMLS:C0149871",
            "label": "Deep Vein Thrombosis",
            "publication": Array [
              "pubmed:18349091",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "UMLS:C0333606",
            "label": "UMLS:C0333606",
            "publication": Array [
              "pubmed:16179904",
              "pubmed:16186368",
              "pubmed:17962476",
              "pubmed:19508456",
              "pubmed:19661213",
              "pubmed:19797200",
              "pubmed:21565171",
              "pubmed:21850171",
              "pubmed:22087103",
              "pubmed:22772592",
              "pubmed:24856391",
              "pubmed:25118264",
              "pubmed:25276414",
              "pubmed:25593508",
              "pubmed:25629076",
              "pubmed:26085992",
              "pubmed:26521715",
              "pubmed:27658286",
              "pubmed:28698241",
              "pubmed:28763560",
              "pubmed:29197825",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "MONDO:0018998",
            "label": "Leber congenital amaurosis",
            "publication": Array [
              "pubmed:28453600",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "UMLS:C0600518",
            "label": "Choroidal Neovascularization",
            "publication": Array [
              "pubmed:21850171",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "UMLS:C0699748",
            "label": "UMLS:C0699748",
            "publication": Array [
              "pubmed:16179904",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "UMLS:C0877373",
            "label": "Advanced cancer",
            "publication": Array [
              "pubmed:20003241",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "MONDO:0000365",
            "label": "primary congenital glaucoma (disease)",
            "publication": Array [
              "pubmed:24856391",
            ],
          },
          Object {
            "api": "SEMMED Gene API",
            "id": "UMLS:C0009566",
            "label": "UMLS:C0009566",
            "publication": Array [
              "pubmed:25056909",
            ],
          },
          Object {
            "api": "BioLink API",
            "id": "MONDO:0008865",
            "label": "Bietti crystalline corneoretinal dystrophy",
            "publication": Array [
              "pubmed:21565171",
              "pubmed:15937078",
              "pubmed:15042513",
              "pubmed:25629076",
              "pubmed:26085992",
              "pubmed:15860296",
              "pubmed:25593508",
              "pubmed:26865810",
              "pubmed:23661369",
              "pubmed:19508456",
              "pubmed:22693542",
              "pubmed:23793346",
              "pubmed:24739949",
              "pubmed:17962476",
            ],
          },
          Object {
            "api": "CTD API",
            "id": "MONDO:0008865",
            "label": "Bietti crystalline corneoretinal dystrophy",
            "publication": Array [],
          },
        ],
        "id": "NCBIGene:285440",
        "label": "cytochrome P450 family 4 subfamily V member 2",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
